rpc/lib/client/http_client.go

```diff
@@ -122,8 +152,41 @@ func (c *JSONRPCClient) Call(method string, params map[string]interface{}, resul
 	if err != nil {
 		return nil, err
 	}
-	// 	log.Info(Fmt("RPC response: %v", string(responseBytes)))
-	return unmarshalResponseBytes(c.cdc, responseBytes, result)
+	return unmarshalResponseBytes(c.cdc, responseBytes, c.id, result)
+}
+
+// NewRequestBatch starts a batch of requests for this client.
+func (c *JSONRPCClient) NewRequestBatch() *JSONRPCRequestBatch {
+	return &JSONRPCRequestBatch{
+		requests: make([]*jsonRPCBufferedRequest, 0),
+		client:   c,
+	}
+}
+
+func (c *JSONRPCClient) sendBatch(requests []*jsonRPCBufferedRequest) ([]interface{}, error) {
+	reqCount := len(requests)
+	reqs := make([]types.RPCRequest, reqCount)
+	results := make([]interface{}, reqCount)
+	for i := 0; i < reqCount; i++ {
```

Comment:
why not use `for range` which is generally safer?

Reply:
For me it's just been a convention: by default I'd use the C-style loop, and when I want to take advantage of the more short-hand 2-tuple variables unpacked by `for range` then I go with that.

Also, over here specifically it's just as safe as the `for range` option, unless `requests` is changing in size, or `len(requests)` comes back with an incorrect value, or `reqs` and/or `results` have been incorrectly allocated by the `make` function - all of which seem impossible to me here (and if those things do occur, then we have much bigger problems 😁).

I can stick to `for range` if that's the style we'd generally prefer.

Reply:
Yeah, I think we want to use `for range` where we can because it provides additional guarantees like the code won't panic with `out of bound` exception